components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -311,68 +312,140 @@ public void deleteUser(String userId) throws NotFoundException, CharonException
+            throw new CharonException(String.format("Error while listing usernames from domain: %s.", domainName), e);
+        }
+    }
 
-                if (isPaginatedUserStoreAvailable()) {
-                    if (carbonUM instanceof PaginatedUserStoreManager) {
-                        User[] scimUsers = this.getSCIMUsers(userNames, requiredClaimsInLocalDialect, scimToLocalClaimsMap);
-                        users.addAll(Arrays.asList(scimUsers));
-                    } else {
-                        retriveSCIMUsers(users, userNames, requiredClaimsInLocalDialect, scimToLocalClaimsMap);
-                    }
-                } else {
-                    retriveSCIMUsers(users, userNames, requiredClaimsInLocalDialect, scimToLocalClaimsMap);
-                }
+    /**
+     * Method to list usernames of all users across all user stores.
+     *
+     * @return Usernames list
+     * @throws CharonException Error while listing usernames
+     */
+    private String[] listAllUsernamesAcrossAllDomains() throws CharonException {
 
-                //set the totalResults value in index 0
-                users.set(0, users.size()-1);
+        String[] userNames = null;
+        try {
+            Map<String, String> scimToLocalClaimsMap = SCIMCommonUtils.getSCIMtoLocalMappings();
+            String userIdLocalClaim = scimToLocalClaimsMap.get(SCIMConstants.CommonSchemaConstants.ID_URI);
+            if (StringUtils.isNotBlank(userIdLocalClaim)) {
+                userNames = carbonUM.getUserList(userIdLocalClaim, SCIMCommonConstants.ANY, null);
             }
+            return userNames;
         } catch (UserStoreException e) {
-            throw new CharonException("Error while retrieving users from user store..", e);
+            throw new CharonException("Error while listing usernames across all domains. ", e);
         }
-        log.info("User list is retrieved through SCIM.");
-        return users;
     }
 
-    private void retriveSCIMUsers(List<Object> users, String[] userNames, List<String> requiredClaims, Map<String,
-            String> scimToLocalClaimsMap)
+    /**
+     * Method to get user details of usernames.
+     *
+     * @param userNames          Array of usernames
+     * @param requiredAttributes Required attributes for the response
+     * @return User list with detailed attributes
+     * @throws CharonException Error while retrieving users
+     */
+    private List<Object> getUserDetails(String[] userNames, Map<String, Boolean> requiredAttributes)
             throws CharonException {
+
+        List<Object> users = new ArrayList<>();
+        try {
+            Map<String, String> scimToLocalClaimsMap = SCIMCommonUtils.getSCIMtoLocalMappings();
+            List<String> requiredClaims = getOnlyRequiredClaims(scimToLocalClaimsMap.keySet(), requiredAttributes);
+            List<String> requiredClaimsInLocalDialect;
+            if (MapUtils.isNotEmpty(scimToLocalClaimsMap)) {
```

Comment:
Can we check `scimToLocalClaimsMap` is empty before retrieving `requiredClaims`?